src/ldap_auth.rs

```diff
@@ -10,7 +10,18 @@ pub fn auth(user: &str, pass: &str, config: &LdapConfig) -> io::Result<bool> {
         return Ok(false);
     }
 
-    let user_filter = format!("({}={})", config.userid_attribute, ldap_escape(user).as_ref());
+    let user_safe = ldap_escape(user);
+    let user_filters = config.userid_attributes.iter().map(|a| format!("({}={})", a, user_safe.as_ref())).collect::<Vec<_>>();
+
+    let user_filter;
+    if user_filters.len() == 0 {
```

Comment:
thought maybe about having some smart defaults here... or maybe just be explicit...?